### ISSUE TITLE
RTCDataChannel.onmessage throws exception when binaryType is "arraybuffer"

### ISSUE THREAD

**Issue by wxyz-abcd** [murat-dogan/node-datachannel#303]
Hi. I am trying to switch from @koush/wrtc to your library. I am using the polyfill in node.js environment. When I try to receive binary data from a RTCDataChannel, in [this line](https://github.com/murat-dogan/node-datachannel/blob/be3860975be2e124057f41cb3ac22c425feddb5f/src/polyfill/RTCDataChannel.ts#L79), I encountered this error:

_First argument to DataView constructor must be an ArrayBuffer_

The argument is received as an `Uint8Array`, but then converted to a node.js `Buffer` by default, while it should in fact take into account the value of `binaryType` before the conversion.

**Comment by wxyz-abcd** [comment]
I added 2 different fixes. Both work perfectly fine for me, but I prefer #305 over #304

**Comment by murat-dogan** [comment]
Hello and thank you for the PRs

The better will be create a jest test for this.
Send a string, blob, buffer and test it like p2p test.

I can make it when I have time, or you can create one if you have time.
Then we can be sure that it will work in all cases and merge one of PRs

what you think?

**Comment by wxyz-abcd** [comment]
Hello, no problem.
I think the second PR is correct, but forget about those PR's. I will create a new much bigger PR soon. I had to change some typescript definitions in order to be able to test the new code, I also added some polyfill test codes and ran them myself and I can confirm that they are running correctly without any problems, at least on my computer. Lastly, according to [this link](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/binaryType), the default value of binaryType is "blob" so I also fixed that within this PR.
